common/djangoapps/util/course.py

```diff
@@ -4,25 +4,24 @@
 import logging
 from django.conf import settings
 
-from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.catalog.utils import get_run_marketing_url
 
 log = logging.getLogger(__name__)
 
 
-def get_lms_link_for_about_page(course_key):
+def get_link_for_about_page(course_key, user, catalog_course_run=None):
     """
     Returns the url to the course about page.
     """
-    assert isinstance(course_key, CourseKey)
-
     if settings.FEATURES.get('ENABLE_MKTG_SITE'):
-        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
-        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
-        about_base = settings.MKTG_URLS['ROOT']
-    else:
-        about_base = settings.LMS_ROOT_URL
+        if catalog_course_run:
```

Comment:
@rlucioni `catalog_course_run` is the cached value for any course; if the cache value is available we just return `marketing_url` from it otherwise we see if catalog integration is enabled then get catalog data else return lms `course_about` url

This condition is one step before CatalogIntegration.